dojo/management/commands/migrate_authorization_v2.py

```diff
@@ -0,0 +1,88 @@
+import logging
+from django.conf import settings
+from django.core.management.base import BaseCommand
+from dojo.models import Dojo_User, Product, Product_Member, Product_Type, Product_Type_Member
+from dojo.authorization.roles_permissions import Roles
+
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    When changing to authorization v2, this command adds staff users and
+    authorized users as product type members / product members according
+    to their permissions they have had before and the settings.
+    """
+    help = 'Usage: manage.py migration_authorization_v2'
+
+    def handle(self, *args, **options):
+
+        logger.info('Started migrating users for authorization v2 ...')
+
+        authorized_user_exists = False
+
+        product_types = Product_Type.objects.all().prefetch_related('authorized_users')
+        staff_users = Dojo_User.objects.filter(is_staff=True, is_superuser=False)
+        for product_type in product_types:
+            # Staff users have had all permissions for all product types and products,
+            # so they will be set as owners for all product types.
+            # Superusers will have all permissions anyway, so they must not be set as members.
+            for staff_user in staff_users:
+                # If the product type member already exists, it won't be changed
+                if Product_Type_Member.objects.filter(product_type=product_type, user=staff_user).count() == 0:
+                    product_type_member = Product_Type_Member()
+                    product_type_member.product_type = product_type
+                    product_type_member.user = staff_user
+                    product_type_member.role = 4  # Owner
+                    product_type_member.save()
+                    logger.info('Product_Type_Member added: {} / {} / {}'.format(product_type.name, staff_user.username, Roles(product_type_member.role).name))
+                else:
+                    logger.info('Product_Type_Member already exists: {} / {}'.format(product_type.name, staff_user.username))
```

Comment:
I don't think we should do this. I think we should have superuser as the global type of users that have all permissions. Or possibly a global owner role talked about. Staff users I think should be purely for sysadmin kinda users, or solely used to grant access to /admin as Django seems to have intended this flag.
Adding staff users as owners to existing product types, but not to new ones might be confusing.

Reply:
Staff users can open the admin interface, but they don't see anything there and they can't do any configuration in DefectDojo.

![2021-05-15 12_54_11-Site administration _ Django site admin](https://user-images.githubusercontent.com/2698502/118357927-ba3c0580-b57c-11eb-97e1-93fab7cf6ad2.png)

For me staff users are employee of my company and non-staff users are external users who need access to DefectDojo.

Superusers can do administration and have full access to the data. This doesn't change with the new authorization.

For staff users there is a change of behaviour with the new authorization. While they had full access to data until now, they will need to be authorized per Product_Type, Product or Group by default in the future. To get the old behaviour, the system parameter `AUTHORIZATION_STAFF_OVERRIDE` can be set to `True`, then all staff users will have full access to all data again. Or a staff user is given a global role (will come soon) and then she will have access to all data according to the role.

So what the behaviour for staff users is with the new authorization is highly configurable but outside of this PR. Here we give staff users the same permissions for the new authorization as they had before.

Reply:
> Staff users can open the admin interface, but they don't see anything there and they can't do any configuration in DefectDojo.
> 
> ![2021-05-15 12_54_11-Site administration _ Django site admin](https://user-images.githubusercontent.com/2698502/118357927-ba3c0580-b57c-11eb-97e1-93fab7cf6ad2.png)
> 
> For me staff users are employee of my company and non-staff users are external users who need access to DefectDojo.

I have never heard that one, don't think that's how it's supposed to work or designed.
 
> Superusers can do administration and have full access to the data. This doesn't change with the new authorization.
> 
> For staff users there is a change of behaviour with the new authorization. While they had full access to data until now, they will need to be authorized per Product_Type, Product or Group by default in the future. To get the old behaviour, the system parameter `AUTHORIZATION_STAFF_OVERRIDE` can be set to `True`, then all staff users will have full access to all data again. Or a staff user is given a global role (will come soon) and then she will have access to all data according to the role.

So with global roles this is no longer needed. To me it makes no sense to add staff users to each product type.

Reply:
Yes, using a global role makes sense. So I convert the PR to draft status until the global roles are implemented.